CMakeLists.txt

```diff
@@ -55,7 +55,23 @@ function(packageProject)
     target_include_directories(
       ${PROJECT_NAME} ${VISIBILITY} "$<BUILD_INTERFACE:${PROJECT_VERSION_INCLUDE_DIR}>"
     )
-    install(DIRECTORY ${PROJECT_VERSION_INCLUDE_DIR}/ DESTINATION ${PROJECT_INCLUDE_DESTINATION})
+    install(
+      DIRECTORY ${PROJECT_VERSION_INCLUDE_DIR}/
+      DESTINATION ${PROJECT_INCLUDE_DESTINATION}
+      COMPONENT "${PROJECT_NAME}_Development"
+    )
+  endif()
+
+  set(sizeof_void_p "${CMAKE_SIZEOF_VOID_P}")
+  if(PROJECT_ARCH_INDEPENDENT)
+    get_target_property(target_type "${PROJECT_NAME}" TYPE)
+    if(NOT TYPE STREQUAL "INTERFACE_LIBRARY")
+      message(
+        WARNING "${PROJECT_NAME} tried to use ARCH_INDEPENDENT, but it is not an INTERFACE library"
+      )
+    else()
+      set(CMAKE_SIZEOF_VOID_P "")
+    endif()
```

Comment:
Interesting, is it documented behaviour that setting `CMAKE_SIZEOF_VOID_P` to an empty string disables the arch check? Would be great if there was some documentation to this behaviour so we can assume that it won't break between CMake versions.

If I understand correctly, the reason we do this is that we currently do not set a minimum CMake version. Do you think it would be better to define the minimum version to `3.14` and pass the `PROJECT_ARCH_INDEPENDENT` to `write_basic_package_version_file` instead?

Reply:
You can see the code [here](https://github.com/Kitware/CMake/blame/master/Modules/BasicConfigVersion-SameMajorVersion.cmake.in#L57) for this. It hasn't changed in over a decade and `ARCH_INDEPENDENT` basically just does the same thing.

With CMake 3.14, this "hack" can be dropped and the namelink can be created with `NAMELINK_COMPONENT` instead, so that would simplify at least 2 things.

Reply:
From the code I agree that it does look like the "intended undocumented" way of doing this.

I do think it should be ok to target CMake 3.14 as a minimum though, as even `ubuntu-latest` on GitHub CI seems to support it, and most users of this project should be able to use a fairly modern toolchain as well. If you agree, I'd prefer to use the clean version of setting a minimum and using the `PROJECT_ARCH_INDEPENDENT` feature as documented.